doc/user/sql-grammar/sql-grammar.bnf

```diff
@@ -264,6 +264,8 @@ type_interval_val ::=
   'INTERVAL' "'" time_expr+ "'" ( ( head_time_unit 'TO' )? tail_time_unit )?
 type_jsonb ::=
     "'" json_string "'" '::JSONB'
+type_map ::=
+    "'" map_string "'" '::MAP[text=>value_type]'
```

Comment:
In the above `value_type` should definitely not be quoted to indicated that it's a variable, similar to `map_string`.

Let's try:

```
"'" map_string "'" '::' 'MAP' '[' 'TEXT' '=>' value_type ']'
```

This breaks everything out into their independently parsed components and hints that `TEXT` could be replaced with other values.

Apologies that the BNF diagrams are inconsistent; we only use them cosmetically, so it's hard to enforce a real structure on them and the standards have drifted over the last year.

Reply:
That totally makes sense, thanks for your help!